### PR TITLE
tree: Improve TreeChangeEvents docs

### PR DESCRIPTION
**Pull request by CraigMacomber**
## Description

 Improve TreeChangeEvents docs.

Split out from https://github.com/microsoft/FluidFramework/pull/22229

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

### CODE REVIEW

File: packages/dds/tree/src/simple-tree/core/types.ts [hunk 1]
```diff
@@ -47,13 +52,13 @@ export type Unhydrated<T> = T;
...
+	 * Emitted by a node after a batch of changes has been applied to the tree, if any of the changes affected the node.
+	 *
+	 * - Object nodes define a change as being when the value of one of its properties changes (i.e., the property's value is set, including when set to `undefined`).
 	 *
-	 * - For an object node, when the value of one of its properties changes (i.e., the property's value is set
-	 * to something else, including `undefined`).
+	 * - Array nodes define a change as when an element is added, removed, moved or replaced.
 	 *
-	 * - For an array node, when an element is added, removed, or moved.
+	 * - Map nodes define a change as when an entry is added, updated, or removed.
```
Comment (alexvy86) [inline packages/dds/tree/src/simple-tree/core/types.ts:61]: The next (non-empty) line is now redundant.

**Review by alexvy86 — APPROVED** [review]
One small thing below.

File: packages/dds/tree/src/simple-tree/core/types.ts [hunk 2]
```diff
@@ -27,6 +27,11 @@ export type Unhydrated<T> = T;
 /**
  * A collection of events that can be emitted by a {@link TreeNode}.
  *
+ * @remarks
+ * Currently events can be subscribed to for {@link Unhydrated} nodes, however no events will be triggered for the nodes until after they are hydrated.
```
Comment (Josmithr) [inline packages/dds/tree/src/simple-tree/core/types.ts:31]: Nit
```suggestion
 * Currently, events can be subscribed to for {@link Unhydrated} nodes, however no events will be triggered for the nodes until after they are hydrated.
```

**Review by Josmithr — APPROVED** [review]

File: packages/dds/tree/src/simple-tree/core/types.ts [hunk 3]
```diff
@@ -47,13 +52,13 @@ export type Unhydrated<T> = T;
...
+	 * - Object nodes define a change as being when the value of one of its properties changes (i.e., the property's value is set, including when set to `undefined`).
 	 *
-	 * - For an object node, when the value of one of its properties changes (i.e., the property's value is set
-	 * to something else, including `undefined`).
+	 * - Array nodes define a change as when an element is added, removed, moved or replaced.
 	 *
-	 * - For an array node, when an element is added, removed, or moved.
+	 * - Map nodes define a change as when an entry is added, updated, or removed.
 	 *
 	 * - For a map node, when an entry is added, updated, or removed.
```
Comment (CraigMacomber) [inline packages/dds/tree/src/simple-tree/core/types.ts:63]:
```suggestion
```

**Pull request merged**